### ISSUE TITLE
[Bug]: v5.0.2 jsoncpp_static, ZXing-cpp

### ISSUE THREAD

**Issue by leukimi** [BelledonneCommunications/linphone-desktop#720]
In order to compile v5.0.2 there are some challenges.

`liblinphone` for some reason requires the use of `jsoncpp_static`, however we have to use the `jsoncpp` shared library. One way to fix this is to compile liblinphone with option `ENABLE_FLEXIAPI=OFF`. Another way is to patch `src/CmakeLists.txt`

Would it be possible to not require the use of `jsoncpp_static` and be happy with the shared library?

`ZXing-cpp` (QR-code) option does not want to compile error free so it has been temporarily disabled. Are you maybe using a special version of `ZXing-cpp` which isn't part of the distributions?

On Ubuntu 22.04 we get this error:

```
[  119s] [ 64%] Building CXX object src/CMakeFiles/linphone.dir/chat/encryption/lime-x3dh-encryption-engine.cpp.o
[  119s] cd /usr/src/packages/BUILD/obj-x86_64-linux-gnu/src && /usr/bin/c++ -DBCTBX_LOG_DOMAIN=\"liblinphone\" -DCXX_WRAPPER_ENABLED -DDEBUG -DHAVE_CONFIG_H -DLINPHONE_EXPORTS -DVCARD_ENABLED -DVIDEO_ENABLED -Dlinphone_EXPORTS -I/usr/src/packages/BUILD/coreapi -I/usr/src/packages/BUILD/include -I/usr/src/packages/BUILD/src -I/usr/src/packages/BUILD/obj-x86_64-linux-gnu -I/usr/src/packages/BUILD/obj-x86_64-linux-gnu/coreapi -I/usr/include/OpenGL -I/usr/src/packages/BUILD/libxsd -I/usr/include/libxml2 -isystem /usr/include/belle-sip -g -O2 -ffile-prefix-map=/usr/src/packages/BUILD=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -g -fPIC -std=c++14  -Wall -Wconversion -Werror=return-type -Winit-self -Wno-error=deprecated-declarations -Wpointer-arith -Wuninitialized -Wunused -Wnon-virtual-dtor -Woverloaded-virtual -Wsuggest-override -Wno-error=suggest-override -MD -MT src/CMakeFiles/linphone.dir/chat/encryption/lime-x3dh-encryption-engine.cpp.o -MF CMakeFiles/linphone.dir/chat/encryption/lime-x3dh-encryption-engine.cpp.o.d -o CMakeFiles/linphone.dir/chat/encryption/lime-x3dh-encryption-engine.cpp.o -c /usr/src/packages/BUILD/src/chat/encryption/lime-x3dh-encryption-engine.cpp
[  122s] /usr/src/packages/BUILD/src/chat/encryption/lime-x3dh-encryption-engine.cpp: In member function 'virtual LinphonePrivate::AbstractChatRoom::SecurityLevel LinphonePrivate::LimeX3dhEncryptionEngine::getSecurityLevel(const std::__cxx11::list<std::__cxx11::basic_string<char> >&) const':
[  122s] /usr/src/packages/BUILD/src/chat/encryption/lime-x3dh-encryption-engine.cpp:645:78: error: cannot convert 'const std::__cxx11::list<std::__cxx11::basic_string<char> >' to 'const string&' {aka 'const std::__cxx11::basic_string<char>&'}
[  122s]   645 |         return limeStatus2ChatRoomSecLevel(limeManager->get_peerDeviceStatus(deviceId));
[  122s]       |                                                                              ^~~~~~~~
[  122s]       |                                                                              |
[  122s]       |                                                                              const std::__cxx11::list<std::__cxx11::basic_string<char> >
```

### General information

- Device: OBS Build Service
- OS:  All Linux OS
- Version of the App: 5.0.2
- Version of the SDK: 5.2.0

### Expected behaviour

Build success.

### To Reproduce

The logs are found using the link to  [Monitor](https://build.opensuse.org/project/monitor/home:kimi:telephony-ithod)

**Comment by julonexus** [comment]
Hi
I don't know why the static was requested. I have to ask.
Is https://gitlab.linphone.org/BC/public/liblinphone/-/commit/e1bdefa370b53ddf31ba8a9b0e70cea3bf950e4b good enough?

**Comment by julonexus** [comment]
Hi
I think it will be fixed for 5.0.3 by just removing this dependency (it is used for an experimental feature) : https://gitlab.linphone.org/BC/public/linphone-desktop/-/commit/74d509cade2ccce97f45f65e9d0df5fb3a8d28af.
Regards
